src/commands/init.ts

```diff
@@ -1,57 +1,117 @@
-import { writeFile } from "fs/promises";
-import { Document } from "yaml";
+import { readFile, writeFile } from "fs/promises";
+import { existsSync } from "fs";
+import { resolve } from "path";
+import { Document, parseDocument } from "yaml";
 import { Config, PluginMap, RuleMap } from "@monokle/validation";
 import { command } from "../utils/command.js";
 import { print } from "../utils/screens.js";
 import { isDefined } from "../utils/isDefined.js";
-import { promptForFrameworks, promptForKubernetesVersion, success, error } from './init.io.js';
+import { promptForFrameworks, promptForKubernetesVersion, promptForOverwrite, success, error } from './init.io.js';
 import { Framework, getFrameworkConfig } from "../frameworks/index.js";
-import { resolve } from "path";
+import defaultConfig from "../utils/defaultConfig.js";
 
-type Options = {};
+type Options = {
+  version?: string;
+  framework?: Framework;
+  overwrite?: boolean;
+};
 
 export const init = command<Options>({
   command: "init",
   describe: "Generate a Monokle policy config file for your project.",
-  async handler() {
+  builder(args) {
+    return args
+      .option("schema-version", {
+        type: "string",
+        description: "Kubernetes version to use.",
+        alias: "v",
+      })
+      .option("framework", {
+        type: "string",
+        choices: ["pss-restricted", "pss-baseline", "nsa"] as const,
+        description: "Validation framework to use.",
+        alias: "f",
+      })
+      .option("overwrite", {
+        type: "boolean",
+        description: "Overwrite existing config file.",
+        default: false,
+        alias: "o",
+      });
+  },
+  async handler({ schemaVersion, framework, overwrite }) {
+    const interactive = !isDefined(schemaVersion) && !isDefined(framework) && isDefined(overwrite);
+    const configPath = resolve(process.cwd(), 'monokle.validation.yaml');
+    const hasConfig = existsSync(configPath);
+
+    if (hasConfig && !interactive && !overwrite) {
+      throw `Config file already exists in ${configPath}. Use --overwrite flag to overwrite it.`
+    }
+
     try {
-      const kubernetesVersion = await promptForKubernetesVersion();
+      let kubernetesVersion = schemaVersion;
+      let frameworks = framework ? [framework] : [];
 
-      if (!isDefined(kubernetesVersion)) {
-        return;
-      }
+      if (interactive) {
+        kubernetesVersion = await promptForKubernetesVersion();
 
-      const frameworks = await promptForFrameworks();
+        if (!isDefined(kubernetesVersion)) {
+          return;
```

Comment:
We have validation there, so empty value only happens when prompt is user cancelled (e.g. `Ctrl+C`). As this is user intention I don't think we should throw here, WDYT?

Hmmm, OTOH we should show some message that init was aborted. Still, maybe regular message instead of error (which indicates that something unexpected happened) could be better? Apart form that, we can add non-zero exit code, but since it's interactive scenario I'm not sure if that would matter? 🤔